spec/migrations/permission_adder_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require Rails.root.join("db/migrate/migration_utils/permission_adder.rb")
+
+RSpec.describe Migration::MigrationUtils::PermissionAdder, type: :model do # rubocop:disable RSpec/SpecFilePathFormat
+  let!(:role) { create(:project_role) }
+  let!(:non_member_role) { create(:non_member) }
+  let!(:anonymous_role) { create(:anonymous_role) }
+
+  let(:roles) { [role, non_member_role, anonymous_role] }
+
+  context "with a permission without special requirements" do
+    it "adds the permission to all roles" do
+      expect(role.permissions).not_to include(:view_project_attributes)
+      expect(non_member_role.permissions).not_to include(:view_project_attributes)
+      expect(anonymous_role.permissions).not_to include(:view_project_attributes)
+
+      described_class.add(:view_project, :view_project_attributes)
+      roles.map(&:reload)
```

Comment:
Nit: Since we are not using the return value, this should be `each`, shouldn't it?

```suggestion
      roles.each(&:reload)
```

Reply:
Copied this over from the old implementation. But makes sense